### PR TITLE
ECC-1825 work in progress

### PR DESCRIPTION
**Pull request by dinamit1984**

### CODE REVIEW

**Review by RobertLaverick — CHANGES_REQUESTED** [review]

File: app/uk/gov/hmrc/eoricommoncomponent/frontend/viewModels/CheckYourDetailsRegisterViewModel.scala [hunk 1]
```diff
@@ -0,0 +1,539 @@
...
+    registration: RegistrationDetails,
+    subscription: SubscriptionDetails,
+    personalDataDisclosureConsent: Boolean,
+    service: Service,
+    isUserIdentifiedByRegService: Boolean
+  )(implicit messages: Messages): CheckYourDetailsRegisterViewModel = {
+
+    val isIndividual: Boolean = cdsOrgType.contains(CdsOrganisationType.Individual) || cdsOrgType.contains(
+      CdsOrganisationType.EUIndividual
+    ) || cdsOrgType.contains(CdsOrganisationType.ThirdCountryIndividual)
```
[the remark this answers is not in the document]
Comment (dinamit1984) [reply app/uk/gov/hmrc/eoricommoncomponent/frontend/viewModels/CheckYourDetailsRegisterViewModel.scala:150]: As Rob suggested  Will raise a spike to investigate how we use our CDS org types

File: test/unit/viewModels/CheckYourDetailsRegisterViewModelSpec.scala [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+
+  private val individualToTest =
+    Seq(
+      Option(CdsOrganisationType.Individual),
+      Option(CdsOrganisationType.EUIndividual),
+      Option(CdsOrganisationType.ThirdCountryIndividual)
+    )
+
+  private val soleTraderToTest =
+    Seq(Some(CdsOrganisationType.SoleTrader), Some(CdsOrganisationType.ThirdCountrySoleTrader))
```
[the remark this answers is not in the document]
Comment (dinamit1984) [reply test/unit/viewModels/CheckYourDetailsRegisterViewModelSpec.scala:66]: As Rob suggested  Will raise a spike to investigate how we use our CDS org types

**Review by burns238 — APPROVED** [review]

File: app/uk/gov/hmrc/eoricommoncomponent/frontend/viewModels/CheckYourDetailsRegisterViewModel.scala [hunk 3]
```diff
@@ -0,0 +1,539 @@
...
+            )
+        }
+      else
+        Seq(
+          summaryListRow(
+            key = businessDetailsLabel(isPartnership, cdsOrgType),
+            value = Some(addressHtml(registration.address)),
+            call = Some(SixLineAddressController.showForm(isInReviewMode = true, organisationType = orgType, service))
+          )
+        )
```
[the remark this answers is not in the document]
Comment (burns238) [reply app/uk/gov/hmrc/eoricommoncomponent/frontend/viewModels/CheckYourDetailsRegisterViewModel.scala]: Something like this @dinamit1984 :

```    
    val (registeredAddress, addressCall) = (isUserIdentifiedByRegService, subscription.addressDetails) match {
      case (false, Some(address)) => 
        (addressViewModelHtml(address), ConfirmContactDetailsController.form(service, isInReviewMode = true))
      case _ => 
        (addressHtml(registration.address), SixLineAddressController.showForm(isInReviewMode = true, organisationType = orgType, service))
    }

    val registeredAddressRow = Seq(
      summaryListRow(
        key = businessDetailsLabel(isPartnership, cdsOrgType),
        value = Some(registeredAddress),
        call = Some(addressCall)
      )
    )```

File: app/uk/gov/hmrc/eoricommoncomponent/frontend/viewModels/CheckYourDetailsRegisterViewModel.scala [hunk 4]
```diff
@@ -0,0 +1,539 @@
...
+      )
+    }).flatten
+
+    val contactName = Seq(subscription.contactDetails.map { cd =>
+      summaryListRow(
+        key = messages("cds.form.check-answers.contact-name"),
+        value = Some(Text(cd.fullName).asHtml),
+        call =
+          Some(uk.gov.hmrc.eoricommoncomponent.frontend.controllers.routes.ContactDetailsController.reviewForm(service))
+      )
```
[the remark this answers is not in the document]
Comment (burns238) [reply app/uk/gov/hmrc/eoricommoncomponent/frontend/viewModels/CheckYourDetailsRegisterViewModel.scala]: This as well ^

**Review by burns238 — CHANGES_REQUESTED** [review]

**Review by burns238 — APPROVED** [review]

**Pull request merged**